app/controllers/sessions_controller.rb

```diff
@@ -26,4 +26,22 @@ def destroy
     log_out if logged_in?
     redirect_to root_url
   end
+
+  def omniauth
+    user = User.find_or_create_by(uid: request.env['omniauth.auth'][:uid],
```

Comment:
Line 31 to 38 should move to model.
Putting logic here will make the controller fat.

Reply:
em đã sửa